py/objlist.c

```diff
@@ -156,7 +156,7 @@ STATIC mp_obj_t list_subscr(mp_obj_t self_in, mp_obj_t index, mp_obj_t value) {
         // delete
 #if MICROPY_PY_BUILTINS_SLICE
         if (MP_OBJ_IS_TYPE(index, &mp_type_slice)) {
-            mp_obj_list_t *self = self_in;
+            mp_obj_list_t *self = MP_OBJ_CAST(self_in);
```

Comment:
> This is my dream :)

Sounds nice for a dream, but this would be really long way. And all efficient things appear to be hand-crafted. Also, isn't PyPy does just that - a description of language (not necessarily Python) which can be parametrized, and for which efficient interpreter/compiler are generated (taking several gigabytes and dozen of minutes in the process).